bears/vcs/mercurial/HgCommitBear.py

```diff
@@ -0,0 +1,210 @@
+import re
+import shutil
+import os
+
+from urllib.parse import urlparse
+from coala_utils.ContextManagers import change_directory
+from coalib.misc.Shell import run_shell_command
+from coalib.results.Result import Result
+from bears.vcs.CommitBear import _CommitBear
+from coalib.settings.FunctionMetadata import FunctionMetadata
+
+
+class HgCommitBear(_CommitBear):
+    LANGUAGES = {'Hg'}
+    SUPPORTED_HOST_KEYWORD_REGEX = {
+        'bitbucket': (r'[Cc]los(?:e[sd]?|ing)'
+                      r'|[Rr]esolv(?:e[sd]?|ing)'
+                      r'|[Ff]ix(?:e[sd]|ing)?')
+    }
+
+    @classmethod
+    def get_shortlog_checks_metadata(cls):
+        return FunctionMetadata.from_function(
+            cls.check_shortlog,
+            omit={'self', 'shortlog'})
+
+    @classmethod
+    def get_body_checks_metadata(cls):
+        return FunctionMetadata.from_function(
+            cls.check_body,
+            omit={'self', 'body'})
+
+    @classmethod
+    def get_issue_checks_metadata(cls):
+        return FunctionMetadata.from_function(
+            cls.check_issue_reference,
+            omit={'self', 'body'})
+
+    @classmethod
+    def get_metadata(cls):
+        return FunctionMetadata.merge(
+            FunctionMetadata.from_function(
+                cls.run,
+                omit={'self', 'dependency_results'}),
+            cls.get_shortlog_checks_metadata(),
+            cls.get_body_checks_metadata(),
+            cls.get_issue_checks_metadata())
+
+    @classmethod
+    def check_prerequisites(cls):
+        if shutil.which('hg') is None:
+            return 'hg is not installed.'
+        else:
+            return True
+
+    @staticmethod
+    def get_host_from_remotes():
+        """
+        Retrieve the first host from the list of mercurial remotes.
+        """
+        remotes, _ = run_shell_command('hg paths')
```

Comment:
is there some parser for these?

e.g. something like https://github.com/coala/coala-bears/issues/2330

Reply:
There exists an npm library for url parsing for bitbucket (https://www.npmjs.com/package/parse-bitbucket-url) specifically, 
Don't know if we have python bindings for it or something similar

Reply:
Create your own equivalent of https://github.com/retr0h/git-url-parse for hg?

It will definitely be useful to other people.
https://github.com/davidfischer/requirements-parser could be enhanced to provide a parsed url.
https://github.com/reinout/checkoutmanager/blob/master/checkoutmanager/config.py#L26 is another one.

Or if the syntax is similar, maybe provide a patch to https://github.com/retr0h/git-url-parse ;-)

Or libvcs or vcstools may be suitable places to put url parsing; I've created https://github.com/vcstools/vcstools/issues/133

Now I think about it, this is the same problem as https://gitlab.com/gitmate/open-source/IGitt/issues/111 , where it needs to determine the hoster, however IGitt doesnt support BitBucket yet ;-)

Reply:
Cool
I will try out building a parser and probably sending an patch :sweat_smile:

Reply:
https://github.com/retr0h/git-url-parse should be very usable for hg.  the syntax appears to be identical, at least as far as we need wrt a parser.  The fixes are not yet released, so test against its master.
We'll help him do a release.

Reply:
https://github.com/coala/coala-bears/pull/2455 introduced a parser which probably will work for this.

Reply:
Seems like the parser in #2455 the parser is unable to support bitbucket hg url's for ssh
We might have to switch to https://github.com/retr0h/git-url-parse